app/models/vm.rb

```diff
@@ -121,6 +121,18 @@ def supported_consoles
     }
   end
 
+  def validate_v2v_migration
+    vm_as_resources = ServiceResource.where(:resource => self).includes(:service_template).select { |rsc| rsc.service_template.kind_of?(ServiceTemplateTransformationPlan) }
```

Comment:
`ServiceResource.where(:resource => self)` Any reason this isn't a relation?

Reply:
Also, any reason not to use `.where(:type => "ServiceTemplateTransformationPlan")` rather than `.select`?